cpp/src/toppra/parametrizer/const_accel.cpp

```diff
@@ -0,0 +1,128 @@
+#include <algorithm>
+#include <cmath>
+#include <cstddef>
+#include <cstdlib>
+#include <stdexcept>
+#include <toppra/parametrizer.hpp>
+#include <toppra/parametrizer/const_accel.hpp>
+#include <toppra/toppra.hpp>
+
+namespace toppra {
+namespace parametrizer {
+
+ConstAccel::ConstAccel(GeometricPathPtr path, const Vector& gridpoints,
+                       const Vector& vsquared)
+    : Parametrizer(path, gridpoints, vsquared) {
+  process_internals();
+}
+
+void ConstAccel::process_internals() {
+  m_ts.resize(m_gridpoints.size());
+  m_us.resize(m_gridpoints.size() - 1);
+  m_ts[0] = 0;
+  for (std::size_t i = 0; i < m_gridpoints.size() - 1; i++) {
+    m_us[i] = 0.5 * (m_vs[i + 1] * m_vs[i + 1] - m_vs[i] * m_vs[i]) /
+              (m_gridpoints[i + 1] - m_gridpoints[i]);
+    m_ts[i + 1] =
+        m_ts[i] + 2 * (m_gridpoints[i + 1] - m_gridpoints[i]) / (m_vs[i + 1] + m_vs[i]);
+  }
+  TOPPRA_LOG_DEBUG("process_internals: ts=" << m_ts.transpose());
+  TOPPRA_LOG_DEBUG("process_internals: us=" << m_us.transpose());
+  TOPPRA_LOG_DEBUG("process_internals: vs=" << m_vs.transpose());
+}
+
+bool ConstAccel::evalParams(const Vector& ts, Vector& ss, Vector& vs,
+                            Vector& us) const {
+  TOPPRA_LOG_DEBUG("evalParams: ts=" << ts.transpose());
+  ss.resize(ts.size());
+  vs.resize(ts.size());
+  us.resize(ts.size());
+  int k_grid = 0;
+  for (std::size_t i = 0; i < ts.size(); i++) {
+    // find k_grid s.t                     m_ts[k_grid] <= ts[i] < m_ts[k_grid + 1], or
+    //      k_grid = 0 if                  ts[i] < m_ts[0], or
+    //      k_grid = m_ts.size() - 2 if    ts[i] > m_ts
+    auto ptr = std::lower_bound(m_ts.data(), m_ts.data() + m_ts.size(), ts[i]);
+    if (ptr == m_ts.data() + m_ts.size()) {
+      k_grid = m_ts.size() - 2;
+    } else if (ptr == m_ts.data()) {
+      k_grid = 0;
+    } else {
+      k_grid = ptr - m_ts.data() - 1;
+    }
```

Comment:
Optional: Isn't this equivalent ?
```suggestion
    assert (m_ts.size() >= 2);
    auto ptr = std::lower_bound(m_ts.data() + 1, m_ts.data() + m_ts.size() - 1, ts[i]);
    k_grid = ptr - m_ts.data() - 1;
```

Reply:
Indeed. Thanks!